src/client/pydaos/pydaos_shim.c

```diff
@@ -63,30 +63,48 @@ struct open_handle {
 		}                                                                                  \
 	} while (0)
 
-static daos_handle_t glob_eq;
-static bool          use_glob_eq;
+/** Global event queue */
+static daos_handle_t eq;
 
 /**
  * Implementations of baseline shim functions
  */
 
+static void
+child_handler(void)
+{
+	int rc;
+
+	rc = daos_reinit();
+	if (rc)
+		D_WARN("daos_reinit() failed in child process %d", rc);
+
+	eq = DAOS_HDL_INVAL;
+	rc = daos_eq_create(&eq);
+	if (rc)
+		DL_ERROR(rc, "Failed to re-create global eq");
+}
+
 static PyObject *
 __shim_handle__daos_init(PyObject *self, PyObject *args)
 {
 	int rc;
 
 	rc = daos_init();
-	if ((rc == 0) && (use_glob_eq == 0)) {
-		d_getenv_bool("PYDAOS_GLOB_EQ", &use_glob_eq);
-		if (use_glob_eq) {
-			int ret;
-
-			ret = daos_eq_create(&glob_eq);
-			if (ret) {
-				DL_ERROR(ret, "Failed to create global eq");
-				use_glob_eq = false;
-			}
-		}
+	if (rc)
+		return PyLong_FromLong(rc);
+
+	rc = daos_eq_create(&eq);
+	if (rc) {
+		DL_ERROR(rc, "Failed to create global eq");
+		daos_fini();
+		return PyLong_FromLong(rc);
+	}
+
+	rc = pthread_atfork(NULL, NULL, &child_handler);
+	if (rc) {
```

Comment:
I am not so familiar with this part of the code and I do not understand why it is not needed in this case to call `daos_fini()`